File: dart.js
/*jshint esversion: 6 */
exports.dart = (arr) => {
  if(arr instanceof Array === true){
    var total = 0;
    var perfect = 0;
    for (var i=0; i < arr.length; i++){
      if(typeof arr[i] !== 'number'){
        throw new Error('wrong data type');
      }
      if (arr[i] < 5 && arr[i] >= 0){
        total += 10;
        perfect ++;
      }
      if (arr[i] <= 10 && arr[i] >= 5){
        total += 5;
      }
      if (perfect === arr.length){
        total += 100;
      }
    }
    return total;
	} else {
    throw new Error('wrong data type');
  }
};

File: python-dart/dart.py
def dart ( arr ):
  score = 0
  perfect = 0
  length = len(arr)
  for x in range (0, length):
    if arr[x] < 5 and arr[x] >= 0:
      score += 10
      perfect += 1
    if arr[x] <= 10 and arr[x] >= 5:
      score += 5
    if perfect == length:
      score += 100

  print score
  print perfect

dart([1,2,3,7,10])